### DISCUSSION TITLE
How to use external DbConnection?

### DISCUSSION THREAD

**Discussion by alewmt** [linq2db/linq2db#3439] [Q&A] [answered]
Hi! I use three data access technologies EF, ADO.NET, linq2db. I need to save all changes in one transaction. I managed to do this for EF and ADO.NET using a shared connection and CommittableTransaction. How can I pass an existing open DbConnection enlisted in a transaction to linq2db?

```
using (var connection = scope.ServiceProvider.GetRequiredService<DbConnection>())
using (var transaction = new CommittableTransaction())
{
    await connection.OpenAsync();
    connection.EnlistTransaction(transaction);

}
```
.net core 6

**Comment by MaceWindu** [comment, marked as answer]
1. automatic enlistment should work with linq2db
2. you can pass connection instance to `DataConnection` directly
